### PR TITLE
MISUV-6433: AddBusinessName Mongo implementation

### PR DESCRIPTION
**Pull request by EdwardDennis**

### CODE REVIEW

File: test/controllers/incomeSources/add/AddBusinessNameControllerSpec.scala [hunk 1]
```diff
@@ -442,26 +455,26 @@ class AddBusinessNameControllerSpec extends TestSupport
...
-        setupMockGetSession(Some("value"))
+        setupMockCreateSession(true)
+        setupMockGetSessionKeyMongoTyped[String](Right(Some("testBusinessName")))
         val validBusinessName: String = "Test Business"
         setupMockAgentAuthRetrievalSuccess(testAgentAuthRetrievalSuccess, withClientPredicate = false)
         setupMockGetIncomeSourceDetails()(businessesAndPropertyIncome)
         val redirectUrl = controllers.incomeSources.add.routes.CheckBusinessDetailsController.showAgent().url
-        val redirect = Redirect(redirectUrl)
-        setupMockSetSession(SessionKeys.businessName, validBusinessName, redirect)
+        setupMockSetSessionKeyMongo(Right(true))
```
Comment (SiddharthAtos) [inline test/controllers/incomeSources/add/AddBusinessNameControllerSpec.scala:464]: verify if the right value was set with -> Mockito verify

File: test/controllers/incomeSources/add/AddBusinessNameControllerSpec.scala [hunk 2]
```diff
@@ -317,10 +325,10 @@ class AddBusinessNameControllerSpec extends TestSupport
           val validBusinessName: String = "Test Business"
           setupMockAuthRetrievalSuccess(BaseTestConstants.testIndividualAuthSuccessWithSaUtrResponse())
           setupMockGetIncomeSourceDetails()(businessesAndPropertyIncome)
-          setupMockGetSession(Some("value"))
+          setupMockCreateSession(true)
+          setupMockGetSessionKeyMongoTyped[String](Right(Some("testBusinessName")))
           val redirectUrl = controllers.incomeSources.add.routes.CheckBusinessDetailsController.show().url
-          val redirect = Redirect(redirectUrl)
-          setupMockSetSession(SessionKeys.businessName, validBusinessName, redirect)
+          setupMockSetSessionKeyMongo(Right(true))
```
Comment (SiddharthAtos) [inline test/controllers/incomeSources/add/AddBusinessNameControllerSpec.scala:331]: verify if the right value was set with -> Mockito verify

File: test/controllers/incomeSources/add/AddBusinessNameControllerSpec.scala [hunk 3]
```diff
@@ -242,19 +249,18 @@ class AddBusinessNameControllerSpec extends TestSupport
         val validBusinessName: String = "Test Business"
         setupMockAgentAuthRetrievalSuccess(testAgentAuthRetrievalSuccess, withClientPredicate = false)
         setupMockGetIncomeSourceDetails()(businessesAndPropertyIncome)
-        setupMockGetSession(Some("value"))
+        setupMockCreateSession(true)
+        setupMockSetSessionKeyMongo(Right(true))
```
Comment (SiddharthAtos) [inline test/controllers/incomeSources/add/AddBusinessNameControllerSpec.scala:253]: verify if the right value was set with -> Mockito verify

File: test/controllers/incomeSources/add/AddBusinessNameControllerSpec.scala [hunk 4]
```diff
@@ -132,18 +135,16 @@ class AddBusinessNameControllerSpec extends TestSupport
         val validBusinessName: String = "Test Business"
         setupMockAuthRetrievalSuccess(BaseTestConstants.testIndividualAuthSuccessWithSaUtrResponse())
         setupMockGetIncomeSourceDetails()(businessesAndPropertyIncome)
-        setupMockGetSession(None)
-        val redirect = Redirect(controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.show(incomeSourceType = SelfEmployment, isAgent = false, isChange = false).url)
-        setupMockSetSession(SessionKeys.businessName, validBusinessName, redirect)
+        setupMockCreateSession(true)
+        setupMockSetSessionKeyMongo(Right(true))
```
Comment (SiddharthAtos) [inline test/controllers/incomeSources/add/AddBusinessNameControllerSpec.scala:139]: verify if the right value was set with -> Mockito verify

File: test/controllers/incomeSources/add/AddBusinessNameControllerSpec.scala (hunk 1 above)
Comment (EdwardDennis) [reply test/controllers/incomeSources/add/AddBusinessNameControllerSpec.scala]: I was unable to get the verify working as it wouldn't match the call, accessed it directly through the mockSessionService instead

**Review by SiddharthAtos — APPROVED** [review]

**Pull request merged**